### PR TITLE
Add missing CLIs for channel upgrade

### PR DESCRIPTION
**Pull request by ljoss17**
Closes: #XXX

## Description

This PR introduces 4 `tx` CLIs for each step of a channel upgrade:

* `hermes tx chan-upgrade-init`
* `hermes tx chan-upgrade-try`
* `hermes tx chan-upgrade-ack`
* `hermes tx chan-upgrade-open`

## Testing

In order to test the CLIs use the following instructions:

NOTE: All CLIs should be called from the `simd-scripts` root

* Install locally simapp from the branch `04-channel-upgrades`
* Clone the forked `simd-scripts` https://github.com/ljoss17/simd-scripts and checkout the branch `luca_joss/channel-upgrade-setup`
* Call `make init`
* Create a channel with `hermes --config network/hermes/config.toml create channel --a-chain ibc1 --b-chain ibc2 --a-port transfer --b-port transfer --new-client-connection --yes`
* (Optional) Confirm the channel is created and the version is `ics20-1` with `hermes --config network/hermes/config.toml query channels --chain ibc1 --verbose`
* Init step for channel upgrade with: `hermes --config network/hermes/config.toml tx chan-upgrade-init --dst-chain ibc1 --dst-channel channel-0 --dst-port transfer --src-chain ibc1 --version "{\"fee_version\":\"ics29-1\",\"app_version\":\"ics20-1\"}" --timeout-time 200000`
* (Optional) Confirm the channel is in `InitUpgrade` state with `hermes --config network/hermes/config.toml query channels --chain ibc1 --verbose`
* Try step for channel upgrade with: `hermes --config network/hermes/config.toml tx chan-upgrade-try --dst-chain ibc2 --dst-connection connection-1 --dst-port transfer --src-chain ibc1 --src-channel channel-0 --src-port transfer --timeout-time 200000 --dst-channel channel-0`
* (Optional) Confirm the counterparty channel is in `TryUpgrade` state with `hermes --config network/hermes/config.toml query channels --chain ibc1 --verbose`
* Ack step for channel upgrade with: `hermes --config network/hermes/config.toml tx chan-upgrade-ack --dst-chain ibc1 --dst-connection connection-1 --dst-port transfer --src-chain ibc2 --src-channel channel-0 --src-port transfer --timeout-time 200000 --dst-channel channel-0`
* (Optional) Confirm the channel is in `Open` state and the version is now ` Version("{\"fee_version\":\"ics29-1\",\"app_version\":\"ics20-1\"}")` with `hermes --config network/hermes/config.toml query channels --chain ibc1 --verbose`
* Open step for counterparty channel end with: `hermes --config network/hermes/config.toml tx chan-upgrade-open --dst-chain ibc2 --dst-connection connection-1 --dst-port transfer --src-chain ibc1 --src-channel channel-0 --src-port transfer --timeout-time 200000 --dst-channel channel-0`
* (Optional) Confirm the counterparty channel end is in `Open` state and the version is now ` Version("{\"fee_version\":\"ics29-1\",\"app_version\":\"ics20-1\"}")` with `hermes --config network/hermes/config.toml query channels --chain ibc1 --verbose`

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer-cli/src/commands/tx/channel.rs [hunk 1]
```diff
@@ -744,26 +746,36 @@ pub struct TxChanUpgradeInitCmd {
         value_name = "TIMEOUT_TIMESTAMP",
         help = "Timestamp that, once it has been surpassed on the originating chain, the upgrade will time out. Required if no timeout height is specified."
     )]
-    timeout_timestamp: Option<Timestamp>,
+    timeout_timestamp: u64,
```
Comment (romac) [inline crates/relayer-cli/src/commands/tx/channel.rs:749]: Why this change?

**Review by romac — APPROVED** [review]
Awesome! Just left a quick question

File: crates/relayer-cli/src/commands/tx/channel.rs (hunk 1 above)
Comment (ljoss17) [reply crates/relayer-cli/src/commands/tx/channel.rs:749]: It was to have a similar timeout as `ft-transfer` where a difference in seconds is given instead of a specific `Timestamp`. Maybe renaming it to `timeout_seconds`, same as the one in `ft-transfer`, would be better.

https://github.com/informalsystems/hermes/blob/2c65075fbd4c24c4b947a633116a46ed0a13712a/crates/relayer-cli/src/commands/tx/channel.rs#L765
